src/core/server/elasticsearch/client/configure_client.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { Logger } from '../../logging';
+import { parseClientOptions, ElasticsearchClientConfig } from './client_config';
+
+export const configureClient = (
+  config: ElasticsearchClientConfig,
+  { logger, scoped = false }: { logger: Logger; scoped?: boolean }
+): Client => {
+  const clientOptions = parseClientOptions(config, scoped);
+  const client = new Client(clientOptions);
+
+  client.on('response', (err, event) => {
+    if (err) {
+      logger.error(`${err.name}: ${err.message}`);
+    } else if (config.logQueries) {
+      const params = event.meta.request.params;
+      logger.debug(
+        `${event.statusCode}\n${params.method} ${params.path}\n${params.querystring?.trim() ?? ''}`,
+        {
+          tags: ['query'],
+        }
+      );
+    }
+  });
```

Comment:
Replacement of what was done here

https://github.com/elastic/kibana/blob/719ff259fcfc21e0b13d908227cd016157aeb19b/src/core/server/elasticsearch/elasticsearch_client_config.ts#L189-L197

Question: now that the consumer got access to `response.warnings`, Do we still want to log warnings from here as it's done in the legacy client, or not?

Reply:
What type of `warnings`  passed via `response.warnings`? If warnings aren't request-specific we need to handle them in a centralized way. Otherwise, all consumers have to duplicate the same logic to deal with warnings.

Reply:
I think warnings are per-request, not something global. @rudolf can you confirm (or infirm) that?

Reply:
The `response.warnings` is the parsed `response.headers.warnings` value, so it is request specific.